File: C++ for Everyone, 2nd Ed, Horstmann/Chapter4/P4-19.cpp
//Factoring of integers.Write a program that asks the user for an integer and then
//prints out all its factors.For example, when the user enters 150, the program should
//print
//2
//3
//5
//5

#include <iostream>

using namespace std;

int main()
{
   cout << "Enter an integer: ";
   int num;
   cin >> num; // 65

   int i = 2;

   while (i <= num) // 2 <= 65 y
   {
      if (num % i == 0) // % = 1
      {
         cout << i << endl;
         num = num / i;
      }
      else
      {
         i++;
      }
   }

   return 0;
}

File: Big-Java-for-Everyone-Late-Objects-Horstmann/Chapter-4-Loops/P4_19.java
/*Write a program that prints a multiplication table, like this:
     1    2   3   4   5   6   7   8   9   10
     2    4   6   8   10  12  14  16  18  20
     3    6   9   12  15  18  21  24  27  30
     . . .
     10   20  30  40  50  60  70  80  90  100*/

public class P4_19 {
    public static void main(String[] args) {
	for (int i = 1; i < 11; i++) {
	    for (int j = 1; j < 11; j++) {
		int product = i * j;
		System.out.printf("%4d", product);
	    }
	    System.out.println();
	}
    }
}
